src/connectors/infinite-hits/connectInfiniteHits.js

```diff
@@ -70,23 +70,43 @@ export default function connectInfiniteHits(renderFn, unmountFn) {
   return (widgetParams = {}) => {
     const { escapeHTML = true, transformItems = items => items } = widgetParams;
     let hitsCache = [];
+    let firstReceivedPage = Infinity;
     let lastReceivedPage = -1;
 
-    const getShowMore = helper => () => helper.nextPage().search();
+    const getShowPrevious = helper => () => {
+      /*
+       * Override the current state without triggering a change event
+       * Avoid updating the browser URL when the user display the previous page
+       */
+      helper
+        .overrideStateWithoutTriggeringChangeEvent({
+          ...helper.state,
+          page: firstReceivedPage - 1,
+        })
+        .search();
+    };
+    const getShowMore = helper => () => {
+      helper.setPage(lastReceivedPage + 1).search();
+    };
 
     return {
       getConfiguration() {
         return escapeHTML ? TAG_PLACEHOLDER : undefined;
       },
 
       init({ instantSearchInstance, helper }) {
+        this.showPrevious = getShowPrevious(helper);
         this.showMore = getShowMore(helper);
+        firstReceivedPage = helper.state.page;
+        lastReceivedPage = helper.state.page;
 
         renderFn(
           {
             hits: hitsCache,
             results: undefined,
+            showPrevious: this.showPrevious,
             showMore: this.showMore,
+            isFirstPage: true,
```

Comment:
We don't know whether or not we are on the last page but we do for the first since it's always 0.